packages/cli-link-assets/src/linkAssets.ts

```diff
@@ -0,0 +1,225 @@
+      );
+    }
+
+    const pbxprojPath = findPbxprojFile(iosProjectInfo);
+    iosPbxprojFilePath = path.join(iosPath, pbxprojPath);
+  }
+
+  if (androidAssetsPath.length === 0 && iosAssetsPath.length === 0) {
+    logger.info(
+      inlineString(
+        `It looks like you haven't configured your assets paths in ${chalk.bold(
+          'react-native.config.js',
+        )} file.
+        To can learn more about ${chalk.bold(
+          'link-assets',
+        )} command visit: ${chalk.underline(
+          'https://github.com/react-native-community/cli/blob/main/packages/cli-link-assets/README.md',
+        )}`,
+      ),
+    );
+    return;
+  }
+
+  const rootPath = ctx.root;
+
+  const fontLinkOptions = fontTypes.reduce(
+    (result: LinkOptionsPerExt, fontType) => {
+      result[fontType] = getLinkOptions('font', androidPath, androidAppName);
+      return result;
+    },
+    {},
+  );
+
+  const imageLinkOptions = imageTypes.reduce(
+    (result: LinkOptionsPerExt, imageType) => {
+      result[imageType] = getLinkOptions('image', androidPath, androidAppName);
+      return result;
+    },
+    {},
+  );
+
+  const audioLinkOptions = audioTypes.reduce(
+    (result: LinkOptionsPerExt, audioType) => {
+      result[audioType] = getLinkOptions('audio', androidPath, androidAppName);
+      return result;
+    },
+    {},
+  );
+
+  const linkOptionsPerExt: LinkOptionsPerExt = {
+    ...fontLinkOptions,
+    ...imageLinkOptions,
+    ...audioLinkOptions,
+  };
+
+  const customLinkOptions = getLinkOptions(
+    'custom',
+    androidPath,
+    androidAppName,
+  );
+
+  const linkPlatformOptions: LinkPlatformOptions[] = [
+    {
+      name: 'Android',
+      enabled: true,
+      platform: 'android',
+      rootPath,
+      assetsPaths: androidAssetsPath,
+      manifest: getManifest(androidPath, 'android'),
+      platformConfig: {
+        exists: !!ctx.project.android,
+        path: androidPath,
+      },
+      cleanAssets: cleanAndroidAssets,
+      copyAssets: copyAndroidAssets,
+      linkOptionsPerExt: {
+        otf: linkOptionsPerExt?.otf?.android,
+        ttf: linkOptionsPerExt?.ttf?.android,
+        png: linkOptionsPerExt?.png?.android,
+        jpg: linkOptionsPerExt?.jpg?.android,
+        gif: linkOptionsPerExt?.gif?.android,
+        mp3: linkOptionsPerExt?.mp3?.android,
+      },
+      otherLinkOptions: customLinkOptions.android,
+    },
+    {
+      name: 'iOS',
+      enabled: true,
+      platform: 'ios',
+      rootPath,
+      assetsPaths: iosAssetsPath,
+      manifest: getManifest(iosPath, 'ios'),
+      platformConfig: {
+        exists: !!ctx.project.ios,
+        path: iosPath,
+        pbxprojFilePath: iosPbxprojFilePath!,
+      },
+      cleanAssets: cleanIOSAssets,
+      copyAssets: copyIOSAssets,
+      linkOptionsPerExt: {
+        otf: linkOptionsPerExt?.otf?.ios,
+        ttf: linkOptionsPerExt?.ttf?.ios,
+        mp3: linkOptionsPerExt?.mp3?.ios,
+      },
```

Comment:
Q: Why are we not specifying image assets here?

Reply:
Good catch @szymonrybczak , in practice it doesn't have difference because we will use `customLinkOptions` instead, but I will add the remaining extensions just to be consistent with Android 👍